pkg/edgeview/src/log-search.go

```diff
@@ -0,0 +1,282 @@
+				continue
+			}
+			ftime := getFileTime(l)
+			if ftime == 0 {
+				continue
+			}
+			if ftime >= t2 && ftime <= t1 {
+				file1 := strings.TrimPrefix(l, "./")
+				gfile := logfiletime{
+					filepath: k + "/" + file1,
+					filesec: ftime,
+				}
+				getfiles = append(getfiles, gfile)
+			}
+		}
+	}
+
+	sort.Slice(getfiles, func(i1, i2 int) bool {
+		return getfiles[i1].filesec < getfiles[i2].filesec
+	})
+
+	return getfiles
+}
+
+func searchLiveLogs(pattern string, now int64, typeStr string, idx *int, logjson bool) {
+	files, err := ioutil.ReadDir("/persist/newlog/collect")
+	if err != nil {
+		fmt.Printf("read /persist/newlog/collect error %v\n", err)
+		return
+	}
+
+	for _, l := range files {
+		if !strings.HasPrefix(l.Name(), typeStr) {
+			continue
+		}
+		file := "/persist/newlog/collect/" + l.Name()
+		searchCurrentLogs(pattern, file, typeStr, now, idx, logjson)
+	}
+}
+
+func searchCurrentLogs(pattern, path, typeStr string, now int64, idx *int, logjson bool) {
+	contents, err := ioutil.ReadFile(path)
+	if err != nil {
+		fmt.Printf("read %s file error: %v\n", path, err)
+	}
+	lines := bytes.SplitAfter(contents, []byte("\n"))
+	var selectlines string
+	for _, l := range lines {
+		if !bytes.Contains(l, []byte(pattern)) {
+			continue
+		}
+		selectlines = selectlines + string(l)
+	}
+	bout := fmt.Sprintf("\n current " + typeStr + " log, -- %v --\n", time.Unix(now, 0).Format(time.RFC3339))
+	printColor(bout, colorRED)
+
+	colorMatch(selectlines, pattern, idx, logjson)
+}
+
+func colorMatch(olines, pattern string, idx *int, logjson bool) {
+	closePipe(true)
+	lines := strings.Split(olines, "\n")
+	if strings.Contains(pattern, "|") {
+		pat := strings.Split(pattern, "|")
+		pattern = strings.TrimSuffix(pat[0], " ")
+	}
+	for i, l := range lines[:len(lines)-1] {
+		if logjson {
+			prettyJSON, err := formatJSON([]byte(l))
+			if err == nil {
+				buff := strings.ReplaceAll(string(prettyJSON), pattern, "\033[0;93m"+pattern+"\033[0m")
+				fmt.Printf(" (%d) %s\n", i+1, buff)
+			}
+		} else {
+			var entry logs.LogEntry
+			var content LogContent
+			var bufStr string
+			_ = json.Unmarshal([]byte(l), &entry)
+			err := json.Unmarshal([]byte(entry.Content), &content)
+			*idx++
+			if err != nil {
+				var tlog string
+				if entry.Timestamp != nil {
+					tlog = time.Unix(entry.Timestamp.Seconds, 0).Format(time.RFC3339)
+				}
+				bufStr = fmt.Sprintf(" -(%d) %s, %s, %s, %v(%d)", *idx, strings.TrimSuffix(entry.Content, "\n"), entry.Severity, entry.Source,
+					tlog, entry.Msgid)
+			} else {
+				bufStr = fmt.Sprintf(" -(%d) %s, %s, %s, %s, %s, %s, %s(%d)",
+					*idx, content.Msg, entry.Severity, entry.Filename, entry.Function, content.Objtype,
+					content.Source, content.Time, entry.Msgid)
+			}
+			buff := strings.ReplaceAll(bufStr, pattern, "\033[0;93m"+pattern+"\033[0m")
+			fmt.Printf("%s\n", buff)
+		}
+		if i%20 == 0 {
+			closePipe(true)
+		}
+	}
+}
+
+func getTimeSec(timeline string, now int64) (int64, int64) {
```

Comment:
Looks like this function is extracting two timestamps. Is that correct? A comment would be helpful.

Reply:
will do.